src/apps/core/submodules/alerts/alerts.js

```diff
@@ -193,55 +193,51 @@ define(function(require) {
 					system: '2',
 					apps: '3'
 				},
-				metadataFormat = self.appFlags.alerts.metadataFormat;
-
-			if (_.isEmpty(data)) {
-				return [];
-			}
+				metadataFormat = self.appFlags.alerts.metadataFormat,
+				getMetadata = function(category, key) {
+					return _.get(
+						metadataFormat.categories,
+						[category, key],
+						_.get(metadataFormat.common, key)
+					);
+				},
+				formatValue = function(type, value) {
+					if (type === 'price') {
+						return monster.util.formatPrice({
+							price: value
+						});
+					}
+					return value;
+				};
 
-			return _.chain(data)
+			return _
+				.chain(data)
 				.filter(function(alert) {
 					return _.has(alert, 'category');
 				})
 				.map(function(alert) {
-					var category = alert.category,
-						metadata = _.reduce(alert.metadata,
-							function(metadataArray, value, key) {
-								var formatData = _.get(
-									metadataFormat.categories,
-									category + '.' + key,
-									_.get(metadataFormat.common, key)
-								);
-
-								if (formatData) {
-									var metadataItem = {
-										key: formatData.i18nKey,
-										value: value
-									};
-
-									switch (formatData.valueType) {
-										case 'price':
-											metadataItem.value = monster.util.formatPrice({
-												price: metadataItem.value
-											});
-											break;
-										default: break;
-									}
-
-									metadataArray.push(metadataItem);
-								}
-
-								return metadataArray;
-							}, []);
-
-					return {
-						id: alert.id,
-						title: alert.title,
-						metadata: metadata,
-						message: alert.message,
-						category: category,
-						clearable: alert.clearable
-					};
+					return _.merge({
+						metadata: _
+							.chain(alert.metadata)
+							.reject(function(value, key) {
```

Comment:
Here `lodash#reject` turns the original metadata object into an array, so the key for the following `lodash#map` function is the array index, instead of the object key.

Reply:
Good catch, I'll apply the reject once the object is transformed into an array.